systemd/emulation.go

```diff
@@ -0,0 +1,150 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package systemd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+)
+
+type emulation struct{}
+
+var errNotImplemented = errors.New("not implemented in emulation mode")
+
+func (s *emulation) DaemonReload() error {
+	return errNotImplemented
+}
+
+func (s *emulation) Enable(service string) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Disable(service string) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Start(service ...string) error {
+	return errNotImplemented
+}
+
+func (s *emulation) StartNoBlock(service ...string) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Stop(service string, timeout time.Duration) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Kill(service, signal, who string) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Restart(service string, timeout time.Duration) error {
+	return errNotImplemented
+}
+
+func (s *emulation) Status(units ...string) ([]*UnitStatus, error) {
+	return nil, errNotImplemented
+}
+
+func (s *emulation) IsEnabled(service string) (bool, error) {
+	return false, errNotImplemented
+}
+
+func (s *emulation) IsActive(service string) (bool, error) {
+	return false, errNotImplemented
+}
+
+func (s *emulation) LogReader(services []string, n int, follow bool) (io.ReadCloser, error) {
+	return nil, errNotImplemented
+}
+
+func (s *emulation) AddMountUnitFile(snapName, revision, what, where, fstype string) (string, error) {
+	mountUnitName, options, err := writeMountUnitFile(snapName, revision, what, where, fstype)
+	if err != nil {
+		return "", err
+	}
+
+	cmd := exec.Command("mount", "-t", fstype, what, where, "-o", strings.Join(options, ","))
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return "", fmt.Errorf("cannot mount %s (%s) at %s in pre-bake mode: %s; %s", what, where, fstype, err, string(out))
+	}
+
+	// cannot call systemd, so manually enable the unit by symlinking into multi-user.target.wants
+	mu := MountUnitPath(where)
+	enableUnitPath := filepath.Join(dirs.SnapServicesDir, "multi-user.target.wants", mountUnitName)
+	if err := osSymlink(mu, enableUnitPath); err != nil {
+		return "", fmt.Errorf("cannot enable mount unit %s: %v", mountUnitName, err)
+	}
+	return mountUnitName, nil
+}
+
+func (s *emulation) RemoveMountUnitFile(mountedDir string) error {
+	unit := MountUnitPath(dirs.StripRootDir(mountedDir))
+	if !osutil.FileExists(unit) {
+		return nil
+	}
+
+	isMounted, err := osutilIsMounted(mountedDir)
+	if err != nil {
+		return err
+	}
+	if isMounted {
+		if output, err := exec.Command("umount", "-d", "-l", mountedDir).CombinedOutput(); err != nil {
```

Comment:
Could use a comment:
```
// use detach-loop and lazy unmount
```
or pass the long options, since they're self explanatory